tests/Console/SelfUpdate/GithubClientTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Console\SelfUpdate;
+
+use PhpCsFixer\Console\SelfUpdate\GithubClient;
+use PhpCsFixer\Tests\TestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Console\SelfUpdate\GithubClient
+ */
+final class GithubClientTest extends TestCase
+{
+    public function testGettingTags(): void
+    {
+        $githubClient = new GithubClient();
+
+        self::assertNotSame([], $githubClient->getTags());
```

Comment:
This relies on the network, I this this should be avoided in tests. Can we stub the network call somehow?

Reply:
Yeah, if we want to test `GithubClient` we should add one layer of abstraction that would be injected into `GithubClient` instance and will serve the purpose of HTTP calls. In basic implementation it can use `file_get_contents()` just like it's done now, but the interface itself will enable mocking the retrieved data and focus test tests only on what `GithubClient` does with it.

Reply:
I don't really want to change how the class is used only for the tests, after all, it is not problematic for years. What about this approach? Making the URL a property will allow not to call network resource (although it is the safest can that can be - CI on GitHub calling GitHub's API).

Reply:
I like your approach, I think it's a good compromise. IMO introducing proper dependency injection will be easier if we setup a DIC someday 🤞